src/components/boolean-search/boolean-table.tsx

```diff
@@ -0,0 +1,307 @@
+              })}
+            </select>
+            <span>To: </span>
+            <select
+              className='stageDropdown'
+              value={row.stageTo.Name || defaultValues.stageTo.Name}
+              onChange={(e) => handleChange(index, 'stageTo', e.target.value)}
+            >
+              {row.toStageOptions?.map((option: any, index: any) => {
+                return <option key={index}>{option.Name}</option>;
+              })}
+            </select>
+          </>
+        ),
+        pattern: (
+          <select
+            value={row.pattern || defaultValues.pattern || ''}
+            onChange={(e) => handleChange(index, 'pattern', e.target.value)}
+          >
+            <option style={{ display: 'none' }} disabled selected value=''></option>
+            {pattern?.map((option: any, index: any) => (
+              <option key={index}>{option.Pattern}</option>
+            ))}
+          </select>
+        ),
+        location: (
+          <select
+            value={row.location || defaultValues.location || ''}
+            onChange={(e) => handleChange(index, 'location', e.target.value)}
+          >
+            <option style={{ display: 'none' }} disabled selected value=''></option>
+            {location?.map((option: any, index: any) => {
+              return (
+                <option key={index}>
+                  {option.Pattern_Location}
+                </option>
+              );
+            })}
+          </select>
+        ),
+        actions: <button onClick={(e) => handleRemoveRow(e, index)}>Remove</button>
+      })),
+    [rows, strength, pattern, location, stageFrom, source]
+  );
+
+  const tableInstance = useTable({
+    columns,
+    data
+  });
+  // Create the table instance
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows: tableRows,
+    prepareRow
+  } = tableInstance;
+
+  return (
+    <div className='form-container'>
+      <table {...getTableProps()} className='boolean-table' cellSpacing='10'>
+        <tbody {...getTableBodyProps()}>
+          {headerGroups.map((headerGroup, key) => (
+            <tr {...headerGroup.getHeaderGroupProps()} key={key}>
+              {headerGroup.headers.map((column, key) => (
+                <th {...column.getHeaderProps()} key={key}>{column.render('Header')}</th>
+              ))}
+            </tr>
+          ))}
+          {tableRows.map((row, index) => {
+            prepareRow(row);
+            return (
+              <tr className={`${activeRow === index ? 'active-row' : ''}`}
+                {...row.getRowProps()} key={index} onClick={(e) => handleActiveRow(e, index)}>
+                {row.cells.map((cell, index) => (
+                  <td {...cell.getCellProps()} key={index}>{cell.render('Cell')}</td>
+                ))}
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+
+export default BooleanTable;
+
+BooleanTable.propTypes = {
```

Comment:
After handling the above comment for adding a type to the component, this can be removed.